src/crimson/osd/pg_backend.cc

```diff
@@ -700,6 +700,40 @@ PGBackend::write_iertr::future<> PGBackend::writefull(
   return seastar::now();
 }
 
+static PGBackend::load_metadata_iertr::future<crimson::osd::ObjectContextRef>
+mocked_load_obc(const auto& soid)
+{
+  return crimson::ct_error::object_corrupted::make();
+}
+
+PGBackend::interruptible_future<> PGBackend::rollback(
+  const SnapSet &ss,
+  ObjectState& os,
+  const OSDOp& osd_op,
+  ceph::os::Transaction& txn,
+  osd_op_params_t& osd_op_params,
+  object_stat_sum_t& delta_stats)
+{
+  assert(os.oi.soid.is_head());
+  logger().debug("{} deleting {} and rolling back to old snap {}",
+                  __func__, os.oi.soid, osd_op.op.snap.snapid);
+  return mocked_load_obc(
+    os.oi.soid
```

Comment:
I'm not sure what's going on here.  We already have the head obc, right?  That's where the passed in SnapSet and ObjectState references live.  Presumably the intention is to resolve the rollback snap target to a specific clone?

Reply:
Yes, I think in future @Matan-B will want to acquire clone's `ObjectContext` here. It requires exposing to `PGBackend` the _obc loading_ parts of `PG`. This PR brings support for just a corner case needed to pass the test.

I will rework the mocking part to better express the intention.

Reply:
Reworked a bit.